CMakeLists.txt

```
@@ -63,8 +64,29 @@ configure_file("${CMAKE_SOURCE_DIR}/cmake/templates/license.h.in" include/licens
 
 include(EthOptions)
 configure_project(TESTS)
+set(LATEST_Z3_VERSION "4.8.12")
+set(LOWEST_Z3_VERSION_SUPPORTED "4.8.0")
+find_package(Z3)
+if (${USE_Z3} AND ${Z3_FOUND})
+  if (${STRICT_Z3_VERSION})
+    if ( NOT ( "${Z3_VERSION_STRING}" VERSION_EQUAL ${LATEST_Z3_VERSION} ) )
+      message(
+        FATAL_ERROR
+        "SMTChecker tests require Z3 ${LATEST_Z3_VERSION} for all tests to pass.\n\
+Build with -DSTRICT_Z3_VERSION=OFF if you want to use an older version. \
+You can also use -DUSE_Z3=OFF to build without Z3. In both cases use --no-smt when running tests."
+            )
+    endif()
+  else()
+    if ( NOT ( "${Z3_VERSION_STRING}" VERSION_GREATER_EQUAL ${LOWEST_Z3_VERSION_SUPPORTED} ) )
```

This could be more concise:
```suggestion
    if ("${Z3_VERSION_STRING}" VERSION_LESS ${LOWEST_Z3_VERSION_SUPPORTED}))
```

---

done